x-pack/platform/plugins/shared/lens/public/visualizations/metric/visualization.tsx

```diff
@@ -555,20 +560,33 @@ export const getMetricVisualization = ({
     if (!datasourceLayer) {
       return { state, savedObjectReferences: [] };
     }
+
     // this should clean up the secondary trend state if in conflict
-    const { isNumeric: isMetricNumeric } = getAccessorType(datasourceLayer, state.metricAccessor);
-    const colorMode = getColorMode(state.secondaryTrend, isMetricNumeric);
-    // if there are no conflicts, it's all persistable as is
-    if (colorMode === state.secondaryTrend?.type) {
-      return { state, savedObjectReferences: [] };
+    const { isNumeric: isPrimaryMetricNumeric } = getAccessorType(
+      datasourceLayer,
+      state.metricAccessor
+    );
+    const { isNumeric: isSecondaryMetricNumeric } = getAccessorType(
+      datasourceLayer,
+      state.secondaryMetricAccessor
+    );
+    const colorMode = getColorMode(state.secondaryTrend, isSecondaryMetricNumeric);
+
+    if (
+      colorMode !== state.secondaryTrend?.type ||
+      isInConflictWithPrimary(state.secondaryTrend, isPrimaryMetricNumeric)
```

Comment:
nit/suggestion: what do you think about calling this function something like `isSecondaryTrendConfigInvalid`? IMHO, it can be easier to follow and understand what we should reset from the state
```suggestion
      isSecondaryTrendConfigInvalid(state.secondaryTrend, isPrimaryMetricNumeric)
```